Production nextjs builds are missing @nx/js

### Current Behavior

This PR introduced automatic transpiling: https://github.com/nrwl/nx/pull/16774

It also added an import for @nx/js, a package with is not automatically added to the package.json file or preferably not needed at all but provided in the compiled helpers

### Expected Behavior

Production nextjs builds should run

### GitHub Repo

_No response_

### Steps to Reproduce

1.


### Nx Report

```shell
Node   : 18.16.0
   OS     : darwin arm64
   pnpm   : 8.4.0
   Hasher : Native
   
   nx                 : 16.1.1
   @nx/js             : 16.1.1
   @nx/jest           : 16.1.1
   @nx/linter         : 16.1.1
   @nx/workspace      : 16.1.1
   @nrwl/workspace    : 14.8.8
   @nx/cypress        : 16.1.1
   @nx/devkit         : 16.1.1
   @nrwl/devkit       : 14.8.8
   @nx/eslint-plugin  : 16.1.1
   @nx/nest           : 16.1.1
   @nx/next           : 16.1.3
   @nx/node           : 16.1.1
   @nx/react          : 16.1.1
   @nx/rollup         : 16.1.1
   @nrwl/tao          : 16.1.1
   @nx/web            : 16.1.1
   @nx/webpack        : 16.1.1
   nx-cloud           : 16.0.5
   typescript         : 5.0.4
   ---------------------------------------
   Community plugins:
   @nx-go/nx-go              : 2.7.0
   @nx-plus/docusaurus       : 14.1.0
   @nx-tools/nx-container    : 4.0.3
   @webpro/nx-tsc            : 0.0.1
   nest-commander-schematics : 3.0.0
   ---------------------------------------
   The following packages should match the installed version of nx
     - @nrwl/workspace@14.8.8
     - @nrwl/devkit@14.8.8
     - @nx/next@16.1.3
     - @nrwl/next@16.1.3
   
   To fix this, run `nx migrate nx@16.1.3`
```


### Failure Logs

```shell
node:internal/modules/cjs/loader:1075                                                                                                                                                                                                     
  const err = new Error(message);                                                                                                                                                                                                         
              ^                                                                                                                                                                                                                           
                                                                                                                                                                                                                                          
Error: Cannot find module '@nx/js'                                                                                                                                                                                                        
Require stack:                                                                                                                                                                                                                            
- /usr/src/app/.nx-helpers/with-nx.js                                                                                                                                                                                                     
- /usr/src/app/.nx-helpers/compiled.js                                                                                                                                                                                                    
- /usr/src/app/next.config.js                                                                                                                                                                                                             
    at Module._resolveFilename (node:internal/modules/cjs/loader:1075:15)                                                                                                                                                                 
    at /usr/src/app/node_modules/.pnpm/next@13.4.2-canary.3_@opentelemetry+api@1.4.1_react-dom@18.2.0_react@18.2.0/node_modules/next/dist/server/require-hook.js:180:36                                                                   
    at Module._load (node:internal/modules/cjs/loader:920:27)                                                                                                                                                                             
    at Module.require (node:internal/modules/cjs/loader:1141:19)                                                                                                                                                                          
    at require (node:internal/modules/cjs/helpers:110:18)                                                                                                                                                                                 
    at Object.<anonymous> (/usr/src/app/.nx-helpers/with-nx.js:10:14)                                                                                                                                                                     
    at Module._compile (node:internal/modules/cjs/loader:1254:14)                                                                                                                                                                         
    at Module._extensions..js (node:internal/modules/cjs/loader:1308:10)                                                                                                                                                                  
    at Module.load (node:internal/modules/cjs/loader:1117:32)                                                                                                                                                                             
    at Module._load (node:internal/modules/cjs/loader:958:12) {                                                                                                                                                                           
  code: 'MODULE_NOT_FOUND',                                                                                                                                                                                                               
  requireStack: [                                                                                                                                                                                                                         
    '/usr/src/app/.nx-helpers/with-nx.js',                                                                                                                                                                                                
    '/usr/src/app/.nx-helpers/compiled.js',                                                                                                                                                                                               
    '/usr/src/app/next.config.js'                                                                                                                                                                                                         
  ]                                                                                                                                                                                                                                       
}                                                                                                                                                                                                                                         
                                                                                                                                                                                                                                          
Node.js v18.16.0
```


### Operating System

- [X] macOS
- [X] Linux
- [ ] Windows
- [ ] Other (Please specify)

### Additional Information

_No response_

--- a/e2e/next/src/next.test.ts
+++ b/e2e/next/src/next.test.ts
@@ -177,6 +177,12 @@ describe('Next.js Applications', () => {
       `dist/packages/${appName}/public/shared/ui/hello.txt`
     );
 
+    // Check that compiled next config does not contain bad imports
+    const nextConfigPath = `dist/packages/${appName}/next.config.js`;
+    expect(nextConfigPath).not.toContain(`require("../`); // missing relative paths
+    expect(nextConfigPath).not.toContain(`require("nx/`); // dev-only packages
+    expect(nextConfigPath).not.toContain(`require("@nx/`); // dev-only packages
+
     // Check that `nx serve <app> --prod` works with previous production build (e.g. `nx build <app>`).
     const prodServePort = 4000;
     const prodServeProcess = await runCommandUntil(

--- a/packages/next/plugins/with-nx.ts
+++ b/packages/next/plugins/with-nx.ts
@@ -8,8 +8,6 @@ import type { NextConfigFn } from '../src/utils/config';
 import type { NextBuildBuilderOptions } from '../src/utils/types';
 import type { DependentBuildableProjectNode } from '@nx/js/src/utils/buildable-libs-utils';
 import type { ProjectGraph, ProjectGraphProjectNode, Target } from '@nx/devkit';
-import { readTsConfigPaths } from '@nx/js';
-import { findAllProjectNodeDependencies } from 'nx/src/utils/project-graph-utils';
 
 export interface WithNxOptions extends NextConfig {
   nx?: {
@@ -222,6 +220,10 @@ function withNx(
       forNextVersion('>=13.1.0', () => {
         if (!graph.dependencies[project]) return;
 
+        const { readTsConfigPaths } = require('@nx/js');
+        const {
+          findAllProjectNodeDependencies,
+        } = require('nx/src/utils/project-graph-utils');
         const paths = readTsConfigPaths();
         const deps = findAllProjectNodeDependencies(project);
         nextConfig.transpilePackages ??= [];

